src/rdkafka_admin.c

```diff
@@ -5300,63 +5456,116 @@ rd_kafka_resp_err_t rd_kafka_AlterUserScramCredentialsResponse_parse(rd_kafka_op
         if (rko_result)
                 rd_kafka_op_destroy(rko_result);
 
-        rd_snprintf(errstr, errstr_size,
-                    "AlterUserScramCredentials response protocol parse failure: %s",
-                    rd_kafka_err2str(reply->rkbuf_err));
+        rd_snprintf(
+            errstr, errstr_size,
+            "AlterUserScramCredentials response protocol parse failure: %s",
+            rd_kafka_err2str(reply->rkbuf_err));
 
         return reply->rkbuf_err;
 }
-rd_kafka_resp_err_t rd_kafka_AlterUserScramCredentials(rd_kafka_t *rk,
-                                        rd_kafka_UserScramCredentialAlteration_t **alterations,
-                                        size_t alteration_cnt,
-                                        const rd_kafka_AdminOptions_t *options,
-                                        rd_kafka_queue_t *rkqu){
-       
+
+void rd_kafka_AlterUserScramCredentials(
+    rd_kafka_t *rk,
+    rd_kafka_UserScramCredentialAlteration_t **alterations,
+    size_t alteration_cnt,
+    const rd_kafka_AdminOptions_t *options,
+    rd_kafka_queue_t *rkqu) {
+
         rd_kafka_op_t *rko;
         size_t i;
-        /* Check Duplicates */
-        if(alteration_cnt > 1){
-                rd_list_t *users = rd_list_new(alteration_cnt,rd_free);
-                for(i=0;i<alteration_cnt;i++){
-                        rd_list_add(users,rd_strdup(alterations[i]->user));
-                }
-                rd_list_sort(users, strcmp_void);
-                if(rd_list_find_duplicate(users,strcmp_void)){
-                        rd_list_destroy(users);
-                        return RD_KAFKA_RESP_ERR_DUPLICATE_RESOURCE;
-                }
-                rd_list_destroy(users);
-                
-        }
-        if(!WITH_SSL){
-                int8_t flag = 0;
-                for(i=0;i<alteration_cnt;i++){
-                        if(alterations[i]->alteration_type == RD_KAFKA_USER_SCRAM_CREDENTIAL_ALTERATION_TYPE_UPSERT){
-                                flag = 1;
+
+        static const struct rd_kafka_admin_worker_cbs cbs = {
+            rd_kafka_AlterUserScramCredentialsRequest,
+            rd_kafka_AlterUserScramCredentialsResponse_parse,
+        };
+
+        rko = rd_kafka_admin_request_op_new(
+            rk, RD_KAFKA_OP_ALTERUSERSCRAMCREDENTIALS,
+            RD_KAFKA_EVENT_ALTERUSERSCRAMCREDENTIALS_RESULT, &cbs, options,
+            rkqu->rkqu_q);
+
+        if (alteration_cnt > 0) {
```

Comment:
We should throw a local Invalid Argument error in case alteration_cnt == 0. What do you think?

Reply:
yeah, without letting if fail on the broker